daemon/cluster/executor/container/controller.go

```diff
@@ -403,9 +416,23 @@ func (r *controller) waitReady(pctx context.Context) error {
 
 	ctnr, err := r.adapter.inspect(ctx)
 	if err != nil {
+		// if we have unknown container, it might just not be started yet and
+		// we should probably wait for it to come online, so pass it
+		//
+		// If we're RUNNING or greater and we get UnknownContainer, it probably
+		// means the container has been deleted from docker. This is an actual
+		// real error and we should return it as such
 		if !isUnknownContainer(err) {
 			return errors.Wrap(err, "inspect container failed")
 		}
+		// access task object through r.Task(), which is concurrency-safe
+		t, tErr := r.Task()
+		if tErr != nil {
+			return errors.Wrap(tErr, "error getting controller task")
+		}
+		if t.Status.State >= api.TaskStateRunning {
```

Comment:
This won't work. The task state is not propagated this deeply and there is no guarantee that this is up to date.

The task state should never be observed in the controller itself. Task state is a function of the method being called. For example, if `Wait` is called, there is an assumption that the task is in the running state.